feat: add cache ttl extension point

igobranco opened the pull request:
## Description

Adds the setting `CMS_CACHE_LIMIT_TTL_MODULE` that should have the
`limit_page_cache_ttl` function that would be called to limit the cache
ttl of a page using business logic.
Closes #7296

<!--
If this is a security issue stop right here and follow our documentation:
http://docs.django-cms.org/en/latest/contributing/development-policies.html#reporting-security-issues
-->

## Related resources

<!--

Add here links to existing issues or conversation from GitHub
or any other resource.
-->

* the issue #7296
* previous PR #7297

## Checklist

<!--
Please check the following items before submitting, otherwise,
your pull request will be closed.

Use 'x' to check each item: [x] I have ...
-->

* [x] I have opened this pull request against ``develop``
* [x] I have added or modified the tests when changing logic
* [x] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on [Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

It will need a test so the interface of the extension point is preserved for the future.
If merged, this new feature would need to go to the docs.
The idea of this PR is to complement the issue.

marksweb commented:
Thank you @igobranco 👍

marksweb commented on cms/cache/page.py:
```diff
@@ -45,30 +45,37 @@ def set_page_cache(response):
+        ttl_list.append(MAX_EXPIRATION_TTL)
+
+        # If defined the `CMS_CACHE_LIMIT_TTL_MODULE` extension point
+        if hasattr(settings, 'CMS_CACHE_LIMIT_TTL_MODULE'):
```
From what I can see, `CMS_CACHE_LIMIT_TTL_MODULE` is new in this PR.

Therefore we'll need some documentation to show what this is expected to be, and how to use it.

I think some detail in here; https://docs.django-cms.org/en/latest/how_to/caching.html?highlight=cache

And also something more brief in the advanced settings; https://docs.django-cms.org/en/latest/reference/configuration.html?highlight=settings#advanced-settings

igobranco replied:
I've included the requested documentation.

igobranco commented:
@marksweb I've renamed the setting to `CMS_CACHE_LIMIT_TTL_CLASS`; included the requested documentation and added new tests. I think with this changes it is ready to a new review.

marksweb commented on docs/reference/configuration.rst:
```diff
@@ -990,6 +990,17 @@ Default value of the ``cache`` attribute of plugins. Should plugins be cached by
+..  setting:: CMS_CACHE_LIMIT_TTL_CLASS
+
+CMS_CACHE_LIMIT_TTL_CLASS
+================
```
I think this line needs to match the length of the title to avoid a syntax error.

igobranco replied:
Fixed.

sampaccoud commented on cms/cache/page.py:
```diff
@@ -45,30 +45,40 @@ def set_page_cache(response):
+            import importlib
+            module_name, class_name = class_name.rsplit(".", 1)
+            LimitPageCacheTTL = getattr(importlib.import_module(module_name), class_name)
+            instance = LimitPageCacheTTL()
```
Why not directly point the setting to a function?

marksweb replied:
Yeah this might be a bit neater as a utility function and it might avoid the local import that way.

igobranco replied:
@sampaccoud  and @marksweb  I've reviewed the PR. Change the extension point to reference a function instead of a class. I've also renamed the extension point has: `CMS_LIMIT_TTL_CACHE_FUNCTION`

igobranco commented:
@marksweb can you review the PR?

igobranco commented:
> One thing I can see though is it needs an entry in the unreleased section of the changelog

Added!
